### DISCUSSION TITLE
BinaryTable boolean column error

### DISCUSSION THREAD

**Discussion by johnmurphyastro** [nom-tam-fits/nom-tam-fits#708] [Q&A] [answered]
Hi @attipaci,
I am looking for sum help creating boolean columns in a BinaryTable. My attempts fail with an exception:

**BinaryTable bt = new BinaryTable();
bt.addColumn(BinaryTable.ColumnDesc.createForScalars(boolean.class));
bt.addRowEntries(true);**

This produces the error:
java.lang.ClassCastException: class java.lang.Boolean cannot be cast to class [Z (java.lang.Boolean and [Z are in module java.base of loader 'bootstrap')
Stack trace:
    nom.tam.fits.BinaryTable.javaToFits1D(BinaryTable.java:3621)
    nom.tam.fits.BinaryTable.addRow(BinaryTable.java:2198)
    nom.tam.fits.TableData.addRowEntries(TableData.java:111)

I am using version 1.20.2
Thanks, John Murphy

**Comment by attipaci** [comment]
Hi @johnmurphyastro,

I think I know what's going in.... `addRowEntries()` boxes the argument, so `true` becomes a `Boolean`, which is then wrapped into a `Boolean[]` array for `addRow()`. However, while other boxes types have the identical association to FITS types as primitives, `Boolean[]` is associated to a FITS logical (`true`/`false`/`null`), whereas `boolean[]` is associated to a FITS bit (`true`/`false` only)... I'll have to think about how the conversion can be made more automatic. In the meantime, you might be stuck with the good old way of

```
  bt.addRow(new boolean[] { true });
```

as a workaround...

**Reply by johnmurphyastro** [reply]
Hi @attipaci ,
Thanks for the very fast reply. However:

**BinaryTable bt = new BinaryTable();
bt.addColumn(BinaryTable.ColumnDesc.createForScalars(boolean.class));
bt.addRow(new Boolean[] { true });**

still fails with:
java.lang.ClassCastException: class java.lang.Boolean cannot be cast to class [Z (java.lang.Boolean and [Z are in module java.base of loader 'bootstrap')

Stack trace:
    nom.tam.fits.BinaryTable.javaToFits1D(BinaryTable.java:3621)
    nom.tam.fits.BinaryTable.addRow(BinaryTable.java:2198)

Regards, John Murphy

**Comment by attipaci** [comment, marked as answer]
Converted to Issue #710
